web-wallet/src/lib/stores/networkStore.js

```diff
@@ -71,6 +43,32 @@ const disconnect = () => network.disconnect();
 /** @type {NetworkStoreServices["getCurrentBlockHeight"]} */
 const getCurrentBlockHeight = () => network.blockHeight;
 
+/** @type {NetworkStoreServices["getNetworkName"]} */
+const getNetworkName = async () => {
+  const nodeInfo = await getNodeInfo();
+
+  if (!nodeInfo) {
+    return "Unknown";
+  }
+
+  switch (nodeInfo?.chainId) {
+    case 1:
+      return "Mainnet";
+    case 2:
+      return "Testnet";
+    case 3:
+      return "Devnet";
+    default:
+      return "Localnet";
+  }
+};
+
+/** @type {NetworkStoreServices["getNodeInfo"]} */
+const getNodeInfo = async () => {
```

Comment:
As the node info won't change once retrieved, you can create a `nodeInfo` variable at module level and here:

```js
if (!nodeInfo) {
  nodeInfo = await network.connect().then(net => net.node.info());
}

return nodeInfo;
```

So that we avoid to make multiple calls.

Reply:
I agree with you in terms of reducing network calls, but I don't think we should set variables from within functions starting with `get`.   I will consider refactoring it, though.

Reply:
It's not that I'm suggesting to make a pure function impure: it was already impure in the first place.